src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsIpc/IpcTransport.cs

```diff
@@ -83,7 +83,7 @@ public static async Task<Stream> ConnectAsync(IpcEndpointConfig config, Cancella
                     ".",
                     config.Address,
                     PipeDirection.InOut,
-                    PipeOptions.None,
+                    PipeOptions.Asynchronous,
```

Comment:
@josalem, FYI, I found this through testing. Since the named pipe client isn't created with the Asynchronous option then async token cancellation doesn't work on (at least) the NamedPipeClientStream.ConnectAsync method, which causes the async method to hang forever.